feat: add CLI flag and environment variable to enable legacy API tokens

benglewis opened the pull request:
Add CLI flag and environment variable to enable legacy API token authentication.

* Add `--enable-legacy-api-token` flag to `label_studio/core/argparser.py`.
* Add `LABEL_STUDIO_ENABLE_LEGACY_API_TOKEN` environment variable to `label_studio/core/settings/base.py`.
* Update `docs/source/guide/start.md` to include the new flag and environment variable.
* Modify `label_studio/server.py` to enable legacy API tokens if the flag or environment variable is set.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/HumanSignal/label-studio/pull/7388?shareId=7eedf20c-b202-4741-be92-09b44e09bc70).

benglewis committed:
feat: add CLI flag and environment variable to enable legacy API tokens

Add CLI flag and environment variable to enable legacy API token authentication.

* Add `--enable-legacy-api-token` flag to `label_studio/core/argparser.py`.
* Add `LABEL_STUDIO_ENABLE_LEGACY_API_TOKEN` environment variable to `label_studio/core/settings/base.py`.
* Update `docs/source/guide/start.md` to include the new flag and environment variable.
* Modify `label_studio/server.py` to enable legacy API tokens if the flag or environment variable is set.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/HumanSignal/label-studio?shareId=XXXX-XXXX-XXXX-XXXX).

benglewis commented:
Fixes #7355

pakelley commented on label_studio/server.py:
```diff
@@ -177,6 +177,11 @@ def _create_user(input_args, config):
     user.active_organization = org
     user.save(update_fields=['active_organization'])
 
+    # Enable legacy API tokens if the flag or environment variable is set
```
Could you instead update this in [organizations.functions.create_organization](https://github.com/HumanSignal/label-studio/blob/050837520e0a33baaf6af47c30b08b14266358cf/label_studio/organizations/functions.py#L7)? That's where we set the token settings to begin with, and I'd like to keep that logic as consolidated as possible 🙂

benglewis replied:
Sure. I will deal with it when I'm next it in front of the computer

benglewis replied:
@pakelley Done

benglewis committed:
Add `--enable-legacy-api-token` flag and environment variable for legacy API token support

* **Argument Parser**
  - Add `--enable-legacy-api-token` flag to the argument parser in `label_studio/core/argparser.py`

* **Settings**
  - Add `LABEL_STUDIO_ENABLE_LEGACY_API_TOKEN` environment variable in `label_studio/core/settings/base.py`
  - Set default value to `False`

* **Documentation**
  - Update `docs/source/guide/start.md` to include `--enable-legacy-api-token` and `LABEL_STUDIO_ENABLE_LEGACY_API_TOKEN`

* **Organization Functions**
  - Update `label_studio/organizations/functions.py` to set `legacy_api_tokens_enabled` based on the flag or environment variable

* **Server Initialization**
  - Pass `enable_legacy_api_token` input argument to `create_organization` in `label_studio/server.py`

benglewis commented:
@pakelley Please feel free to make any edits you like if you want to change anything - since I am Out-of-Office for the next couple days

pakelley committed:
minor tidying

pakelley committed:
lint

pakelley committed:
one more doc update

benglewis committed:
Merge branch 'develop' into add-legacy-token-flag

benglewis commented:
@pakelley I fixed merge conflicts merging with `main`, but I see that this now requires you to re-add the Pull Request to the _auto-merge_ queue

pakelley commented:
@benglewis I'm still seeing this in the tests:
```
Run pipx install "poetry==${POETRY_VERSION}"
Unable to parse package spec: poetry==
```
Mind checking again that this is set? For what it's worth, we're cutting a new release tomorrow and I'd love to get this in before the cut 🙂

benglewis commented:
@pakelley I believe that in this repository, the GitHub Actions workflows are not designed to work for people who are not HumanSignal employees. Instead, I saw that with my co-worker ( @ddishi ) what @bmartel did with his code was to create a parallel Pull Request (https://github.com/HumanSignal/label-studio/pull/7000) with the same code which he synchronized and used that to check the GitHub Actions workflows. Hopefully this makes it possible to merge this 🙏

pakelley commented:
I see, thanks for your patience here @benglewis ! I've created #7413 . I'll figure out how we can either make this process easier or at least add it to CONTRIBUTING.md for now.

benglewis commented:
@pakelley Great. Thank you 🙌 . I see that you merged the parallel PR - is it possible to merge this PR too?

pakelley commented:
@benglewis I'm closing this one since the commits from this PR were all merged as part of #7413

micaelakaplan commented:
Hi @benglewis ! Could you email me regarding this PR? micaela@humansignal.com thanks!
